instrumentation/trilogy/lib/opentelemetry/instrumentation/trilogy/patches/client.rb

```diff
@@ -26,6 +26,8 @@ def initialize(options = {})
           end
 
           def ping(...)
+            return super unless OpenTelemetry::Trace.current_span.context.valid? || config[:ping_enabled]
```

Comment:
Is this something that we can mixin during initialization instead of checking every time this method is invoked?